src/js/utils/ui.js

```diff
@@ -21,6 +21,12 @@ const LONG_PRESS_DELAY = 500;
 let longPressTimeout;
 let lastInteractionListener;
 
+if (OS.iOS && OS.version.major < 11) {
+    const body = self.document.body;
```

Comment:
where does this `self` keyword get instantiated? is this something that works on ios 10? something that is set in webpack? has self been `this` all along?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/self

In hls.js we replaced all references to `window` with `self` and this is one of the changes I want to make in our projects with the headless player. https://github.com/jwplayer/jwplayer-commercial/pull/7252

Using `self` as the global object works in node and in browsers both in the main event loop and in workers.

Reply:
the more you know - now the `window.top === window.self` check in analytics makes more sense, thanks!